k2/python/k2/fsa.py

```diff
@@ -193,25 +193,34 @@ def __init__(self,
         _ = self.properties
 
     def __str__(self) -> str:
-        '''Return a string representation of this object (note: does not
-           contain all the information in it for now)'''
+        '''Return a string representation of this object
+
+        For visualization and debug only.
+        '''
         if hasattr(self, 'aux_labels'):
             aux_labels = self.aux_labels.to(torch.int32)
         else:
             aux_labels = None
         if self.arcs.num_axes() == 2:
-            ans = "k2.Fsa: " + _fsa_to_str(self.arcs, False, aux_labels)
+            ans = 'k2.Fsa: ' + _fsa_to_str(self.arcs, False, aux_labels)
         else:
-            ans = "k2.FsaVec: \n"
+            ans = 'k2.FsaVec: \n'
             for i in range(self.shape[0]):
                 # get the i-th Fsa
                 ragged_arc, start = self.arcs.index(0, i)
                 end = start + ragged_arc.values().shape[0]
-                ans += "FsaVec[" + str(i) + "]: " + _fsa_to_str(
+                ans += 'FsaVec[' + str(i) + ']: ' + _fsa_to_str(
                     ragged_arc, False,
                     None if aux_labels is None else aux_labels[start:end])
-        ans += "properties_str = " + _k2.fsa_properties_as_str(
-            self._properties) + "."
+        ans += 'properties_str = ' + _k2.fsa_properties_as_str(
+            self._properties) + '.'
+        for name, value in self.named_tensor_attr(include_scores=False):
+            sep = '\n'
+            ans += f'{sep}{name}: {value}'
+        for name, value in self.named_non_tensor_attr():
+            sep = '\n'
+            ans += f'{sep}{name}: {value}'
```

Comment:
mm I'm concerned if we us symbol tables a lot, this could be too verbose.

Reply:
Ok, I will remove the symbol table from the output.